### ISSUE TITLE
"Disabling the check" in documentation should include in-line exclusion rules

### ISSUE THREAD

**Issue by luksi1** [rodjek/puppet-lint#837]
In the documentation, in-line exclusion rules should be included under "Disabling the check". Otherwise, it's a bit of guesswork on how to disable it.

**Comment by kg4zow** [comment]
I don't see any "voting" buttons, so ... +1

**Comment by usev6** [comment]
@luksi1 @kg4zow  Out of curiosity: Do you have a specific piece of documentation in mind where the information about in-line exclusion rules should be added? (I'm not sure what '"Disabling the check" in documentation' refers to.)

Personally, I was happy with https://github.com/rodjek/puppet-lint#disable-checks-within-puppet-code and the linked http://puppet-lint.com/controlcomments/.

**Comment by luksi1** [comment]
@usev6 Take for example the following code that you linked to regarding disabling checks within puppet-code, but I'll remove the in-line ignore for the line that's longer than 140 character.

```puppet
class foo {
  $bar = 'bar'

  # This ignores the double_quoted_strings check over multiple lines

  # lint:ignore:double_quoted_strings
  $baz = "baz"
  $gronk = "gronk"
  # lint:endignore

  # This ignores the 140chars check on a single line
  $this_line_has_a_really_long_name_and_value_that_is_much_longer_than_the_style_guide_recommends = "I mean, a really, really long line like you can't believe"

}````

This will result in:

`WARNING: line has more than 140 characters on line 12`

The question then becomes: what is the in-line ignore statement supposed to be?  lint:ignore:140_chars? lint:ignore:140_characters? It would be nice when you google the warning, you'd get the documentation for the exact in-line syntax. Instead, you get this:

`puppet-lint --no-140chars-check path/to/file.pp`

The actual syntax for the ignore statement is "lint:ignore:140chars" which is guessable in this case, but there are other examples that are more difficult. Instead though of the documentation on GitHub, as well as puppet-lint.com (this issue is more focused on puppet-lint.com), being strictly focused on the cli exclusions, the documentation should also include the in-line exclusion rules as well, such as "lint:ignore:<rule>".

Hopefully, this clearer.

**Comment by usev6** [comment]
@luksi1 Yes, thanks, I understand the objective of this issue better now. IMHO you made some valid points.

One idea that might help a bit: Maybe it would be useful to add  the check name after each warning/error by default, like this:
```
WARNING: line has more than 140 characters on line 12 (check: 140chars)
```

(There is the option --log-format to do this, but the default doesn't include the name of the check. Maybe a change for the default format would be ok for version 3?)

**Comment by luksi1** [comment]
This would be a perfect solution. Thanks.
